ui/src/pages/publications/[id]/index.tsx

```diff
@@ -77,19 +77,25 @@ export const getServerSideProps: Types.GetServerSideProps = async (context) => {
 type Props = {
     publication: Interfaces.Publication;
     publicationId: string;
-    bookmark: boolean;
+    bookmarkId: string | null;
     userToken: string;
 };
 
 const Publication: Types.NextPage<Props> = (props): React.ReactElement => {
     const router = useRouter();
     const confirmation = Contexts.useConfirmationModal();
-    const [isBookmarked, setIsBookmarked] = React.useState(props.bookmark ? true : false);
+    const isBookmarked = props.bookmarkId ? true : false;
+    const [showBookmark, setShowBookmark] = React.useState(isBookmarked);
+    const [bookmarkId, setBookmarkId] = React.useState(props.bookmarkId);
```

Comment:
```suggestion
    const [bookmarkId, setBookmarkId] = React.useState(props.bookmarkId);
    const isBookmarked = bookmarkId ? true : false;
```
There's a small bug with the bookmarking mechanism while bookmarking a publication and navigating to a new one, the new publication shows as bookmarked as well until the page is refreshed. 
There's another small bug when trying to remove a bookmark, sometimes the bookmarkId is null because the state was not updated.

You may not need 'showBookmark' state since you already have 'bookmarkId' state. We just need to make sure that 'bookmarkId' is updated properly.